packages/wallet/src/redux/protocols/advance-channel/reducer.ts

```diff
@@ -147,57 +145,175 @@ function initializeWithNewChannel(
       sharedData,
     };
   } else {
-    throw new Error('Unimplemented');
+    const protocolState = states.channelUnknown({
+      ...initializeChannelArgs,
+      processId,
+    });
+
+    return { protocolState, sharedData };
   }
 }
 
-function initializeWithExistingChannel(channel, processId, sharedData) {
-  const { ourIndex, channelId } = channel;
-  return { protocolState: states.notSafeToSend({ processId, channelId, ourIndex }), sharedData };
+const channelUnknownReducer: ProtocolReducer<states.AdvanceChannelState> = (
+  protocolState: states.ChannelUnknown,
+  sharedData,
+  action: CommitmentsReceived,
+) => {
+  const { ourIndex, privateKey, commitmentType } = protocolState;
+  const channelId = getChannelId(action.signedCommitments[0].commitment);
+
+  const checkResult = checkAndInitialize(sharedData, action.signedCommitments[0], privateKey);
+  if (!checkResult.isSuccess) {
+    throw new Error('Could not initialize channel');
+  }
+  sharedData = checkResult.store;
+  sharedData = checkCommitments(sharedData, 0, action.signedCommitments);
+  let channel = getChannel(sharedData.channelStore, channelId);
+  if (!channel) {
```

Comment:
Can we remove this check if we have sufficient unit testing?

Reply:
Unfortunately not, without refactoring `getChannel` -- it can return `undefined`.

@lalexgap do you think we could throw in `getChannel` if it doesn't exist?

Reply:
I think that would make sense.